IDinsight/aaq-core
# AAQ-272: Error codes

Reviewer: @amiraliemami 

Estimate: 1 hr

---

## Ticket

Fixes: [AAQ-272](https://idinsight.atlassian.net/browse/AAQ-272)

## Description, Motivation and Context

### Goal

We were piggybacking off the standard response when there was an error.
This was not a clean solution and causes issues for consuming applications (like Botpress).

In the PR, we return a 400 error and different JSON if any of the rails fail.

**Note: This is only a refactor. I'll fix `paraphrase` in another PR.**

### Changes

- New schema - `UserQueryResponseError` that captures the errors
- New table to store errors
- New HTTP return code (400) for these errors
- New tests to make sure these are handled correctly
- Old test updated included `rails` ones

## How has this been tested?

Reminder to run: `python -m alembic upgrade head`

1. Use tests
2. Use SwaggerUI to send a query that cannot be paraphrased - the default text "query" - or language cannot be identified (e.g. "asfsdfsdfgaaw"). Note that HTTP 400 is returned and the error reason specifies why it failed
3. Repeat for both endpoint. 
4. Repeat with a valid query and should get HTTP 200 and a valid response

## Checklist

Fill with `x` for completed. Delete any lines that are not relevant



[AAQ-272]: https://idinsight.atlassian.net/browse/AAQ-272?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
